shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -263,49 +368,75 @@ private void setSystemChromeSystemUIOverlayStyle(
-    // in Android, but you can't change the color of the navigation buttons until
-    // Android O.
-    // LIGHT vs DARK effectively isn't supported until then.
-    // Build.VERSION_CODES.O
-    if (Build.VERSION.SDK_INT >= 26) {
-      if (systemChromeStyle.systemNavigationBarIconBrightness != null) {
-        switch (systemChromeStyle.systemNavigationBarIconBrightness) {
-          case DARK:
-            // View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
-            flags |= 0x10;
-            break;
-          case LIGHT:
-            flags &= ~0x10;
-            break;
-        }
+
+    // SYSTEM STATUS BAR -------------------------------------------------------------------
+    // You can't change the color of the system status bar until SDK 21, but you can't change the
+    // color of the status icons until SDK 23. We only allow both starting at 23 to ensure buttons
+    // and icons can be visible when changing the background color.
+    // If transparent, SDK 29 and higher may apply a translucent scrim behind the bar to ensure
+    // proper contrast. This can be overridden with
+    // SystemChromeStyle.systemStatusBarContrastEnforced.
+    if (systemChromeStyle.statusBarIconBrightness != null && Build.VERSION.SDK_INT >= 23) {
+      switch (systemChromeStyle.statusBarIconBrightness) {
+        case DARK:
+          // View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
+          flags |= 0x2000;
+          break;
+        case LIGHT:
+          flags &= ~0x2000;
+          break;
       }
-      if (systemChromeStyle.systemNavigationBarColor != null) {
-        window.setNavigationBarColor(systemChromeStyle.systemNavigationBarColor);
+
+      if (systemChromeStyle.statusBarColor != null) {
+        window.setStatusBarColor(systemChromeStyle.statusBarColor);
       }
     }
-    // Build.VERSION_CODES.M
-    if (Build.VERSION.SDK_INT >= 23) {
-      if (systemChromeStyle.statusBarIconBrightness != null) {
-        switch (systemChromeStyle.statusBarIconBrightness) {
-          case DARK:
-            // View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
-            flags |= 0x2000;
-            break;
-          case LIGHT:
-            flags &= ~0x2000;
-            break;
-        }
+    // You can't override the enforced contrast for a transparent status bar until SDK 29.
+    // This overrides the translucent scrim that may be placed behind the bar on SDK 29+ to ensure
+    // contrast is appropriate when using full screen layout modes like Edge to Edge.
+    if (!systemChromeStyle.systemStatusBarContrastEnforced && Build.VERSION.SDK_INT >= 29) {
+      window.setStatusBarContrastEnforced(systemChromeStyle.systemStatusBarContrastEnforced);
+    }
+
+    // SYSTEM NAVIGATION BAR --------------------------------------------------------------
+    // You can't change the color of the system navigation bar until SDK 21, but you can't change
+    // the color of the navigation buttons until SDK 26. We only allow both starting at 23 to
```

Comment:
Don't you mean `We only allow both starting at 26...`?  That's what the if statement below is saying.

Reply:
Whoops! Copy pasta. 😋  Thanks!

Reply:
🕵️